Add considerations for multi instance of destinations (#6259)
segmentio/segment-docs: src/connections/destinations/add-destination.md

You can also connect multiple instances of a destination to help you smoothly migrate from one configuration to another. By sending each version the same data, you can check and validate the new configuration without interrupting use of the old one.

> However, there are a few considerations:

Reviewer: hi @courtneyga - these are all good points and would make a useful addition to the docs! 

I do have a few questions, though: 
- How can a user tell if a destination can connect multiple instances to the same source? Is there an attribute in the partner portal that the docs team could use to build some autogenerated info? 
- What is a hybrid actions destination?
- Does the restriction about Data Lakes hold true for AWS, Azure, and Databricks data lakes?

Thanks!

> Some destinations do not support having multiple instances connected to the same source. In that case, the option to add a second instance of that destination does not appear.

Reviewer: ```suggestion
Device-mode destinations do not support connecting multiple instances of the destination to the same source. If you try to a connect an additional instance of a device-mode destination to your source, the option to add a second instance does not appear.
```

> Mobile sources, and the legacy Project source, can connect to multiple instances of destinations that operate only in cloud-mode. Mobile and Project sources cannot connect to multiple instances of destinations that operate in both cloud-mode and device-mode.

Reviewer: ```suggestion
Mobile sources, and the legacy Project source, can connect to multiple instances of destinations that operate only in cloud-mode. Mobile and Project sources cannot connect to multiple instances of destinations that operate in both cloud-mode and device-mode. Non-mobile sources can only connect to _one_ device-mode instance of a destination. 
```

> Multi-instance support is not available for most hybrid Actions destinations, and will not support Web Mode Actions destinations.

Reviewer: ```suggestion
Multi-instance support is not available for most hybrid Actions destinations or Web mode Actions destinations.
```

> Segment does not currently support connecting a single source to multiple instances of a data lakes destination.

Reviewer: ```suggestion
Segment does not support connecting a single source to multiple instances of a [Data Lakes](/docs/connections/storage/data-lakes/) destination.
```
